### PR TITLE
[DEV-1096] Improve release processes

### PR DESCRIPTION
**Pull request by anikitinDSR**
Signed-off-by: Andrew Nikitin <andrew.nikitin@cheqd.io>

### CODE REVIEW

**Comment by ankurdotb** [comment]
Task linked: [DEV-1096 Create release candidate branches for April On Chain Upgrade + document processes](https://app.clickup.com/t/6600954/DEV-1096)

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -6,104 +6,136 @@ on:
...
-          PREFIX: "/usr/local"
-          VERSION: "1.0.0-rc8"
-        run: |
-          curl -sSL "https://github.com/bufbuild/buf/releases/download/v${VERSION}/buf-$(uname -s)-$(uname -m).tar.gz" | \
-          sudo tar -xvzf - -C "${PREFIX}" --strip-components 1
-
+          ref: ${{ github.sha }}
+          running-workflow-name: 'Wait for test and lint workflows'
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
+          wait-interval: 30
```
Comment (askolesov) [inline .github/workflows/release.yml:19]: Should we decrease the interval to 10 seconds?

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -17,21 +16,21 @@ jobs:
...
-    steps:
-      - name: Checkout code
-        uses: actions/checkout@v2
-
-      - name: Check if Markdown links are valid
-        uses: gaurav-nelson/github-action-markdown-link-check@v1
-        with: 
-          config-file: '.github/linters/mlc_config.json'
-          folder-path: "."
+#  md-link-check:
```
Comment (askolesov) [inline .github/workflows/lint.yml:19]: We can't leave it commented

File: build-tools/build-deb.sh [hunk 3]
```diff
@@ -7,7 +7,14 @@ if [ -z ${1+x} ]; then
 fi
 
 BINARY_PATH="$1"
-VERSION=$("${BINARY_PATH}" version 2>&1)
+# It's needed for creating an RC deb package
```
Comment (askolesov) [inline build-tools/build-deb.sh:10]: My heart bleeds when I realize that versions of a binary and the deb package it belongs can be different.

**Review by askolesov — APPROVED** [review]
Thanks, @anikitinDSR! Great work, a lot of improvements, new cool solutions. Code looks clean. The only thing - I would discuss it with @ankurdotb. If he is OK with this approach, let's merge this PR.

File: .github/workflows/lint.yml (hunk 2 above)
Comment (askolesov) [reply .github/workflows/lint.yml:19]: Oh, and we should solve this before merging

Comment (anikitinDSR) [reply .github/workflows/lint.yml:19]: As I can see, for now links to images are working. I'll get back this check

File: .github/workflows/release.yml (hunk 1 above)
Comment (anikitinDSR) [reply .github/workflows/release.yml:19]: Firstly, it was 10 seconds by default but I increased it because of lags on starting for our workflows.

**Review by ankurdotb — APPROVED** [review]

**Pull request merged**